mlir/lib/Dialect/MIOpen/Transforms/ThreadwiseToStdlib.cpp

```diff
@@ -1524,11 +1526,16 @@ struct XdlopsGemmV2RewritePattern : public OpRewritePattern<XdlopsGemmV2Op> {
               ValueRange{offset});
         }
       } else {
-        auto constructArg = [&outerLoopb, &innerLoopb, &loc,
-                             &b](Value &buffer, Value &arg, Value &outerLoopiv,
-                                 Value &innerLoopiv, Type &argType,
-                                 int64_t k_base) {
-          arg = createZeroConstantOp(innerLoopb, loc, argType);
+        // This lambda create an xdlops argument from the vgpr buffer.
+        // Each time in the nested loop, it first loads a vector of size
+        // kpack guarateed to be a multiple of the size of the xdlops
+        // argument. Then it construct the xdlops argument by extracting
+        // elements from the kpack vector.
+        auto constructXdlopsArg = [&outerLoopb, &innerLoopb, &loc,
+                                   &b](Value &buffer, Value &outerLoopiv,
```

Comment:
nitpick: `b` is not really used in this lambda it seems?

Reply:
Nice catch!